draft-ietf-ace-oscore-profile.xml

```diff
@@ -460,44 +439,89 @@ OSCORE_Security_Context = {
+                This parameter identifies the OSCORE Master Secret value, which is a byte string. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>.
+                In JSON, the "ms" value is a Base64 encoded byte string.
+                In CBOR, the "ms" type is bstr, and has label 1.
               </t>
 
               <t hangText="clientId:">
-                This parameter is used to carry a client identifier as a byte string. This identifier is used as OSCORE Sender ID in the client and OSCORE Recipient ID in the server. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>.
+                This parameter identifies a client identifier as a byte string. This identifier is used as OSCORE Sender ID in the client and OSCORE Recipient ID in the server. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>.
+                In JSON, the "clientID" value is a Base64 encoded byte string.
+                In CBOR, the "clientID" type is bstr, and has label 2.
               </t>
 
               <t hangText="serverId:">
-                This parameter is used to carry a server identifier as a byte string. This identifier is used as OSCORE Sender ID in the client and OSCORE Recipient ID in the server. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>.
+                This parameter identifies a server identifier as a byte string. This identifier is used as OSCORE Sender ID in the client and OSCORE Recipient ID in the server. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>.
+                In JSON, the "serverID" value is a Base64 encoded byte string.
+                In CBOR, the "serverID" type is bstr, and has label 3.
               </t>
               
               <t hangText="hkdf:">
-                This parameter is used to carry the OSCORE HKDF Algorithm, which can be an integer or a text string.
-                The int value MUST be taken and can only encode an alg from the set of HMAC-based KDF algorithms in the COSE Algorithms registry.
-                For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>
+                This parameter identifies the OSCORE HKDF Algorithm. For more information about this field, see section 3.1 of <xref target="I-D.ietf-core-object-security"/>                
+                The values used should either be registered in the IANA "COSE Algorithms" registry, or be a value that contains a name allocated in a way that is highly unlikely to collide with other names. From the "COSE Algorithms" registry, only HMAC-based HKDF algorithms MUST be used.
```

Comment:
I don't want to have random values here that don't match COSE algorithms.  I would prefer not to have the second option.

Reply:
Actually, that was my oversight, since OSCORE specifies: "The HKDF MUST be one of the HMAC-based HKDF [RFC5869] algorithms defined for COSE [RFC8152]." So yes only COSE algorithms can be used... but they could still be the full name rather than the registered id. Is that ok?